### DISCUSSION TITLE
Flashforge SGVX: File does not contain a resolution and/or display size information needed to generate the layer images

### DISCUSSION THREAD

**Discussion by doyundoyun** [sn4k3/UVtools#463] [Q&A] [answered]
```
System.IO.FileLoadException: This file does not contain a resolution and/or display size information needed to generate the layer images.
Note that FlashDLPrint slicer is unable to output files with the required information to load in here.
Please use other compatible slicer capable of output the correct information to load the file in here.
File name: 'C:\Users\doyun\OneDrive\Desktop\20mm_Box.svgx'
at UVtools.Core.FileFormats.FlashForgeSVGXFile.DecodeInternally(OperationProgress progress) in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.Core\FileFormats\FlashForgeSVGXFile.cs:line 640
at UVtools.Core.FileFormats.FileFormat.Decode(String fileFullPath, FileDecodeType fileDecodeType, OperationProgress progress) in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.Core\FileFormats\FileFormat.cs:line 3324
at UVtools.WPF.MainWindow.<>c__DisplayClass104_0.b__1() in D:\Tiago\Dropbox\Programming\C#\UVtools\UVtools.WPF\MainWindow.axaml.cs:line 1453
```

![image](https://user-images.githubusercontent.com/90031064/163716201-f783e674-0c88-4d08-b325-6ac53a807cab.png)

http://gofile.me/4zQ31/kh0vSGhGS

Please download and look into this SVGX file.

**Comment by sn4k3** [comment, marked as answer]
The error dialog already tell you the problem and solution...

Quote:
This file does **not contain** a **resolution** and/or **display size information** needed to generate the layer images.
Note that **FlashDLPrint slicer is unable to output files with the required information to load in here.**
Please use **other compatible slicer** capable of output the correct information to load the file in here.

SVGX is just an vector file, lacking of any other precious information for UVtools to work. 
UVtools works with images and those require Resolution size to be known, without it UVtools can't recreate the image from the svgx file, and is not possible to just guess it.

So this is not a bug, but rather a incompability by flashforge and the way they built the format. 
The only solution to use UVtools with SVGX is using other slicer that is able to convert or output svgx with my required meta data inserted.

- PrusaSlicer (With my printer profiles)

Is the only slicer able to slice and export svgx (after open sl1 on UVtools) with inserted metadata for UVtools to work (I altered the svgx format with the missing information that is required)

But you can use other slicer as well, like Chitubox or Lychee, Tango, which i highly recommend over flashforge slicer, create a printer with same dimentions and properties, export to a known format, like zip, load into UVtools, do your stuff and convert to SVGX. 


Also SVGX files created by UVtools also insert the missing information, so you can load it back if required

**Comment by doyundoyun** [comment]
Hi, Tiago.

Thanks for your quick response.

I have tried both with Tango and Prusaslicer to create svgx but it did not
work.

Could you please advise where to insert the resolution and display size
information in Flashforge's svgx file?

Resolution : 1920*1080 pixels
Display size : 120*67.5*150 mm (4.7*2.7*5.9 INCHES)

I have attached a 20mm box sliced svgx file.
Thank you very much.

Warm regards,
David Lee

**Reply by sn4k3** [reply]
Prusaslicer is unable to creat svgx, you need to export sl1 and open in uvtools with my printer profiles and it will turn into .svgx automaticly: https://github.com/sn4k3/UVtools/wiki/Setup-PrusaSlicer

Tango can do the same thing, you export to vdt and then go to uvtools and convert to SVGX.
Any of these are better than insert the bytes yourself

![2022-04-18_16-15-18](https://user-images.githubusercontent.com/113281/163829824-023ffbf9-2e7a-449a-94ab-ae94571d232b.png)


> Could you please advise where to insert the resolution and display size information in Flashforge's svgx file?

You will need an hex editor able to paste and insert text in right position, heres where (The orange text and replace by your values):

![2022-04-18_16-13-38](https://user-images.githubusercontent.com/113281/163829719-a2704135-b2ba-4fe8-b64d-8de4ca832a7d.png)

**Comment by doyundoyun** [comment]
Dear, Tiago

Thank you.

Everything is perfect now.

How can I support you? do you have a donation link of this software?

Kind regards,

Doyun


2022년 4월 19일 (화) 오전 12:16, Tiago Conceição ***@***.***>님이 작성:

**Reply by sn4k3** [reply]
Yes, on UVtools you can go to Help menu and click either sponsor or donate.
Official sponsor channel it's the github itself: https://github.com/sponsors/sn4k3 (Prefered)

**Reply by mderosa89** [reply]
i am having a similar problem and i am brand new to d printers

System.IO.InvalidDataException: Unable to detect the printer model from resolution, check if resolution is well defined on slicer for your printer model.
   at UVtools.Core.FileFormats.CXDLPFile.EncodeInternally(OperationProgress progress) in UVtools.Core/FileFormats/CXDLPFile.cs:line 667
   at UVtools.Core.FileFormats.FileFormat.Encode(String fileFullPath, OperationProgress progress) in UVtools.Core/FileFormats/FileFormat.cs:line 3551
   at UVtools.Core.FileFormats.FileFormat.Convert(Type to, String fileFullPath, UInt32 version, OperationProgress progress) in UVtools.Core/FileFormats/FileFormat.cs:line 5129
   at UVtools.Core.FileFormats.FileFormat.Convert(FileFormat to, String fileFullPath, UInt32 version, OperationProgress progress) in UVtools.Core/FileFormats/FileFormat.cs:line 5143
   at UVtools.WPF.MainWindow.<>c__DisplayClass113_0.<ConvertToOnTapped>b__0() in UVtools.WPF/MainWindow.axaml.cs:line 1834



I NEED HELP

**Reply by sn4k3** [reply]
@mderosa89 Wrong resolution defined on slicer. Check your settings. Error is clear.
